fhir-server/src/main/java/au/csiro/pathling/QueryHelpers.java

```diff
@@ -272,32 +272,39 @@ private static boolean hasColumn(@Nonnull final Dataset<Row> dataset,
     Dataset<Row> dataset = left.getDataset();
     final List<FhirPath> joinTargets = fhirPaths.subList(1, fhirPaths.size());
 
+    boolean nonLiteralEncountered = left instanceof NonLiteralPath;
```

Comment:
I think this works but is somewhat obscure. Maybe it would clearer if we just filter all NonLiterals up front and either apply the existing logic if there are any or return `fhirPaths.get(0)` if there are none.

Reply:
Fixed in 59295803b97509c48ae9620dc3f13809c425b80a.